designs/static-capacity.md

```diff
@@ -0,0 +1,202 @@
+        name: myEc2NodeClass
+      expireAfter: 720h
+      taints:
+        - key: example.com/special-taint
+          effect: NoSchedule
+      requirements:
+        - key: topology.kubernetes.io/zone
+          operator: In
+          values: ["us-west-2a", "us-west-2b", "us-west-2c"]
+        - key: karpenter.k8s.aws/instance-type
+          operator: In
+          values: ["m5.2xlarge"]
+  disruption:
+    budgets:
+      - nodes: 10%
+      # On Weekdays during business hours, don't do any deprovisioning.
+      - schedule: "0 9 * * mon-fri"
+        duration: 8h
+        nodes: "0"
+  limits:
+    # Constrain the size of static Nodepool 
+    nodes: 20 # Karpenter will not scale up nodes or create replacements if limits are breaching
+```
+
+We will support scaling static node pools using a command such as:
+
+```sh
+kubectl scale nodepool <name> --replicas=10
+```
+
+## Implementation Details
+
+### Controller Architecture
+
+We will be creating Static Capacity by adding 2 new controllers under a feature flag. This will be added to v1alpha1 release to get feedback from users.
+
+#### Scale-Up Controller
+- **Purpose**: Scale up nodeclaims to meet desired replicas
+- **Behavior**: Monitors the difference between desired replicas and current healthy NodeClaims, provisioning new nodes as needed respecting limits.nodes value
+- **Instance Selection**: Will find the cheapest instance possible during scale up operations based on the requirements
+- **Consolidation**: Static NodePools will not be consolidated to maintain predictable capacity
+
+#### Deprovisioning Controller  
+- **Purpose**: Scale down replicas when desired count is reduced
+- **Node Selection**: Will prioritize empty nodes for termination. In future iterations, we will add intelligent algorithms for selecting which nodes to scale down
+- **Termination Behavior**: Will forcefully terminate nodes to meet desired size without respecting PodDisruptionBudgets or terminationGracePeriod
+
+**Important Note**: The scale-down of replicas represents scaling down capacity, not disruption. We don't want to end up in a situation where a user sets the replicas to a lower number and we would still be running significantly more capacity due to blocking pods, PDBs, and terminationGracePeriod constraints.
+
+### Scheduling Integration
+
+We will consider static nodeclaims that are already launched and to be launched during scheduling simulation. This approach will:
+- Lessen the churn and possibility of over-provisioning
+- Be accounted for during disruption/consolidation operations
+- Be considered during new dynamic capacity provisioning decisions
+
+This ensures that the scheduler is aware of both existing and planned static capacity when making provisioning decisions.
+
+### Testing Strategy
+
+For testing, we will add comprehensive integration tests to ensure the feature works correctly across different scenarios:
+- Scale-up operations with various instance types and requirements
```

Comment:
If we're just creating standard NodeClaims I don't think there's any specific testing we need for static capacity. The NodeClaim lifecycle controller's behavior shouldn't differ for NodeClaims created by a dynamic or static nodepool. I think the only thing we really need to verify is that the created nodeclaims adhere to the topology skew requirements.